src/components/Modal/Modal.tsx

```diff
@@ -0,0 +1,88 @@
+import { useForm } from 'react-hook-form';
+import { DataInfo } from '../../types/DataInfo';
+import { ModalTypes } from '../../types/ModalTypes';
+
+import styles from './Modal.module.scss';
+
+export const Modal: React.FC<ModalTypes> = ({ activeModal, setActiveModal, list, setList }) => {
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+    reset,
+  } = useForm<DataInfo>({
+    mode: 'onBlur',
+  });
+
+  const onSubmit = (data: DataInfo) => {
+    let newUser = { ...data, id: (Math.random() + 1).toString(36).substring(2) };
+    setList([...list, newUser]);
+    setActiveModal(false);
+    reset();
+  };
+
+  return (
+    <div
+      className={activeModal ? `${styles.modal} ${styles.active}` : styles.modal}
```

Comment:
`isModalActive`
now it looks like you have a lot of modals and pass the active one